### ISSUE TITLE
Source code is not work correctly.

### ISSUE THREAD

**Issue by tommysnu** [Rich-Hall/sentinel1decoder#5]
Dear Rich-Hall.
First of all, thank you for your source code. However, when I run your code with the Sentinel1A Level0 Folder [https://drive.google.com/file/d/1DBijEmVCP3qK1yR44wnEMIZb21hgdR81/view?usp=sharing](Link)
It cannot give me correct image as bellow. Can you help me check again.
Thank you so much. 
![Image](https://github.com/Rich-Hall/sentinel1decoder/assets/121787065/314e6e31-b4a0-4668-b5d8-864bbb4cee86)

**Comment by Rich-Hall** [comment]
Could you share some details of the data you are using? The example I have used has the satellite in basic Stripmap mode. Gneerally, the satellite operates in [TOPSAR mode](https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-1-sar/products-algorithms/level-1-algorithms/topsar-processing). This electronically steers the beam backward to forward while cycling between different sub-swaths. A lot of additional processing is required to process images acquired in this mode, detailed in the [algorithm definition](https://sentinel.esa.int/documents/247904/1877131/Sentinel-1-Level-1-Detailed-Algorithm-Definition). Unfortunately I have not attempted to process data from this mode, I have stuck to the more basic Stripmap mode (with the beam in a fixed position relative to the satellite) in order to keep the processing simple.

**Comment by tommysnu** [comment]
Yes, I am using the satellite in IW mode. However, after processing, the image I got and the orginal image is difference. Did you have any experience with the IW mode? Thank you so much.

**Comment by tommysnu** [comment]
I got the data from this website and download L0RawData. However the result after processing is difference with orginal image. 
Result ![image](https://github.com/Rich-Hall/sentinel1decoder/assets/121787065/0c6d2271-3699-4c4d-ad71-94468526a79f)
Orginal 
![image](https://github.com/Rich-Hall/sentinel1decoder/assets/121787065/ceeb068a-79b1-41fb-ba8a-21926e0fcd01)
